engine.py

```diff
@@ -990,3 +969,31 @@ def close_windows(self):
 
         # Clear the dictionary of Maya panels now that they were deleted.
         self._maya_panel_dict = {}
+
+    @classmethod
+    def maya_scene_path(cls):
```

Comment:
Please document this method as an internal API and that should not be used from outside the repository. **We** can use it because we package everything together in the engine, but clients shouldn't expect that interface to stay. I'll bring this up with @robblau . That's a clever use of the fact that the publisher hooks live with the engine and we can share code this way between hooks, but we don't want to lock ourselves into supporting that API for everyone yet.

@thebeeland @robblau @willis102 ?

Reply:
Also a little bit of background about why I did this. We converted the Maya engine from using pymel, and during that conversion process, I read the pymel code for getting the scene path. That warned of certain scenarios of where the cmds method wouldn’t return the scene path even if a scene was open. I decided to port that method over to our engine code to keep the functionality the same since I don’t know if that issue still exists.
On porting the publish hooks, I needed to adjust the code for getting the scene name as it was trying to handle Unicode, and I thought it would make sense to make the engine method public and use that instead.

Reply:
Also we do a similar thing in After Effects, and Photoshop, where we provide public methods for common operations. Although admittedly interfacing with Photoshop and After Effects is not so simple, so it makes more sense to wrap common functionality. 
Save method defined in the engine:
https://github.com/shotgunsoftware/tk-aftereffects/blob/master/engine.py#L352
Used in the publish hooks:
https://github.com/shotgunsoftware/tk-aftereffects/blob/master/hooks/tk-multi-publish2/basic/publish_document.py#L303 (